lib/peer.js

```diff
@@ -45,6 +46,22 @@ exports.createTCPIncomingPeer = conn => {
   return peer
 }
 
+/**
+ * Incoming uTP peers start out connected, because the remote peer connected to the
+ * listening port of the uTP server. Until the remote peer sends a handshake, we don't
+ * know what swarm the connection is intended for.
+ */
+exports.createUTPIncomingPeer = conn => {
+  const addr = `${conn.remoteAddress}:${conn.remotePort}`
+  const peer = new Peer(addr, 'utpIncoming')
+  peer.conn = conn
+  peer.addr = addr
+
+  peer.onConnect()
+
+  return peer
+}
+
```

Comment:
seems like you could make these into one function to be more DRY but not important.

Reply:
Added a helper function to improve this 👍